src/lexer.coffee

```diff
@@ -136,6 +136,12 @@ exports.Lexer = class Lexer
     if id is 'default' and @seenExport and @tag() in ['EXPORT', 'AS']
       @token 'DEFAULT', id
       return id.length
+    if id is 'do' and regExSuper = /^(\s*super)(?!\(\))/.exec @chunk[3...]
+      @token 'SUPER', 'super'
+      @token 'CALL_START', '('      
+      @token 'CALL_END', ')'
+      [input, sup] = regExSuper
+      return sup.length + 3
```

Comment:
Yes, 3 is the length of `do` plus a space. 
The `\s*` in `\s*super` was added to catch all spaces between `do` and `super`.
So, when `do` is detected, the regex will match zero or more spaces before `super`, starting from third characters of current match e.g. `do␣`.

If you change line 144 in `lexer` to `return 8` it will work for one space, but will fail when you add more spaces, e.g. `do␣␣␣␣super`.